src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void toStorage(FruitTransaction transaction) {
```

Comment:
before saving the result you must check whether there are enough units in stock, otherwise discard the exception